File: 2025/01/02/urldomain.py
import urllib.parse

def main():
    url = "https://microsoft.coⅯ"

    # Parse the URL
    try:
        parsed_url = urllib.parse.urlparse(url)
        host = parsed_url.netloc  # Equivalent to host in this context
    except ValueError:
        print("URL could not be parsed.")
        return

    # Convert host to UTF-32
    utf32_bytes = bytearray()
    for char in host:
        # Convert each character to its UTF-32 representation
        code_point = ord(char)
        utf32_bytes.extend(code_point.to_bytes(4, byteorder='big'))

    # Convert to hexadecimal for display
    hex_utf32 = utf32_bytes.hex()

    print(f"UTF-32 encoding of the domain: {hex_utf32}")
    print("Note: UTF-32 encoding results in 4 bytes per character, so each character will appear as 8 hexadecimal digits.")

if __name__ == "__main__":
    main()

File: 2025/01/02/urldomain.go
package main

import (
    "fmt"
    "net/url"
)

func main() {
    urlString := "https://microsoft.coⅯ"

    // Parse the URL
    parsedURL, err := url.Parse(urlString)
    if err != nil {
        fmt.Println("URL could not be parsed:", err)
        return
    }

    // Get the host part which is the domain
    host := parsedURL.Host

    // Convert host to UTF-32
    utf32Bytes := make([]byte, 0)
    for _, r := range host {
        // Convert each rune to its 32-bit representation
        utf32Bytes = append(utf32Bytes, byte(r>>24), byte(r>>16), byte(r>>8), byte(r))
    }

    // Convert to hexadecimal for display
    var hexUtf32 string
    for _, b := range utf32Bytes {
        hexUtf32 += fmt.Sprintf("%02x", b)
    }

    fmt.Println("UTF-32 encoding of the domain:", hexUtf32)
    fmt.Println("Note: UTF-32 encoding results in 4 bytes per character, so each character will appear as 8 hexadecimal digits.")
}
